### PR TITLE
Create Unit of Work for handling transactions

### PR DESCRIPTION
**Pull request by GhiocelAndrei**

### CODE REVIEW

File: IssueTrackerApplication/Services/TransactionUnit.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Transactions;
+
+namespace IssueTracker.Application.Services
+{
+    public class TransactionUnit : ITransactionUnit
```
Comment (cristi-paraschiv) [inline IssueTrackerApplication/Services/TransactionUnit.cs:5]: UnitOfWork

Comment (GhiocelAndrei) [reply IssueTrackerApplication/Services/TransactionUnit.cs]: renamed TransactionUnit to UnitOfWork through an Amend

File: IssueTrackerApplication/Services/UnitOfWork.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Transactions;
+
+namespace IssueTracker.Application.Services
+{
+    public class UnitOfWork : IUnitOfWork
+    {
+        public async Task ExecuteWithTransactionAsync(Func<Task> operation)
```
Comment (florincatrina) [inline IssueTrackerApplication/Services/UnitOfWork.cs:7]: Simplicity always wins but at least expose transaction scope param for some flexibility and a return type for the operation

File: IssueTrackerApplication/Services/UnitOfWork.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
+﻿using System.Transactions;
```
Comment (florincatrina) [inline IssueTrackerApplication/Services/UnitOfWork.cs:1]: Omg so much duplication in this class, call the most generic method don't copy paste

File: IssueTrackerApplication/Services/UnitOfWork.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+        public async Task ExecuteWithTransactionAsync(Func<Task> operation, TransactionScopeOption scopeOption)
+        {
+            await ExecuteWithTransactionAsync(operation, scopeOption, new TransactionOptions());
+        }
+
+        public async Task<T> ExecuteWithTransactionAsync<T>(Func<Task<T>> operation, TransactionScopeOption scopeOption)
+        {
+            return await ExecuteWithTransactionAsync(operation, scopeOption, new TransactionOptions());
+        }
```
Comment (florincatrina) [inline IssueTrackerApplication/Services/UnitOfWork.cs:25]: Reduce async await

File: IssueTrackerApplication/Services/IUnitOfWork.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+        Task<T> ExecuteWithTransactionAsync<T>(Func<Task<T>> operation);
+
+        Task ExecuteWithTransactionAsync(Func<Task> operation, TransactionScopeOption scopeOption);
+
+        Task<T> ExecuteWithTransactionAsync<T>(Func<Task<T>> operation, TransactionScopeOption scopeOption);
+
+        Task ExecuteWithTransactionAsync(Func<Task> operation, TransactionScopeOption scopeOption, TransactionOptions transaction);
+
+        Task<T> ExecuteWithTransactionAsync<T>(Func<Task<T>> operation, TransactionScopeOption scopeOption, TransactionOptions transaction);
```
Comment (florincatrina) [inline IssueTrackerApplication/Services/IUnitOfWork.cs:17]: I think only these two need to stay in the interface. The first 4 can be extension methods

**Review by florincatrina — APPROVED** [review]

File: IssueTrackerApplication/Services/UnitOfWork.cs (hunk 3 above)
Comment (GhiocelAndrei) [reply IssueTrackerApplication/Services/UnitOfWork.cs:1]: Solved.

File: IssueTrackerApplication/Services/UnitOfWork.cs (hunk 4 above)
Comment (GhiocelAndrei) [reply IssueTrackerApplication/Services/UnitOfWork.cs]: Solved

File: IssueTrackerApplication/Services/IUnitOfWork.cs (hunk 5 above)
Comment (GhiocelAndrei) [reply IssueTrackerApplication/Services/IUnitOfWork.cs:9]: I created UnitOfWorkExtensions class to solved this issue

**Review by cristi-paraschiv — APPROVED** [review]

**Pull request merged**